src/sdc11073/consumer/consumerimpl.py

```diff
@@ -340,7 +343,10 @@ def base_url(self) -> str:
         if self._http_server is None:
             return ''
         p = urlparse(self._http_server.base_url)
-        tmp = f'{p.scheme}://{self._network_adapter.ip}:{p.port}{p.path}'
+        if self._alternative_hostname:
```

Comment:
can be reduced to `tmp = f'{p.scheme}://{self._alternative_hostname or self._network_adapter.ip}:{p.port}{p.path}'`

Reply:
will be done in next pr